music/demos/piano_genie.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Tone from 'tone';
+import * as mm from '../src/index';
+import {CHECKPOINTS_DIR} from './common';
+
+// tslint:disable-next-line:max-line-length
+const GENIE_CHECKPOINT = `${CHECKPOINTS_DIR}/piano_genie/model/epiano/stp_iq_auto_contour_dt_166006`;
+const LOWEST_PIANO_KEY_MIDI_NOTE = 21;
+
+const genie = new mm.PianoGenie(GENIE_CHECKPOINT);
+
+function initControlsAndAudio () {
+  const buttonToNote = new Map<number, number>();
+  const synth = new Tone.PolySynth(8, Tone.FMSynth).toMaster();
+
+  // Bind keyboard controls
+  document.onkeydown = (evt: KeyboardEvent) => {
+    if (Tone.context.state !== 'running') {
```

Comment:
`Tone.context` gives access to the underlying `AudioContext` object (rather than the tone transport), which does have the 'running' flag.

https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/resume#Example